### PR TITLE
Support canI for sidebar menu

### PR DESCRIPTION
**Pull request by benjaminapetersen**
@jwforres @spadgett 
- Add `canI` support to menu items built via `constants.js`
- Separates out the `canI` filters into their own file.

### CODE REVIEW

File: app/scripts/directives/nav.js [hunk 1]
```diff
@@ -42,6 +43,15 @@ angular.module('openshiftConsole')
           }
           return "project/" + $scope.projectName + href;
         };
+
+        $scope.show = function(item) {
+          var isValid = !item.isValid || item.isValid();
```
Comment (benjaminapetersen) [inline app/scripts/directives/nav.js]: I didn't figure these will be complex/expensive checks but can definitely switch this up.

**Comment by benjaminapetersen** [comment]
So on the initial page load that `show()` function will run:
- 300 times on membership, routes, deployments, etc w/nothing in the project
- 700 times on overview w/ no projects
- 1600 times on overview with the cakePHP sample project
- 350 times on builds with a single build

This isn't necessarily a re-render, but its definitely a signal that there are a lot of $digest loops running.

File: app/scripts/directives/nav.js [hunk 2]
```diff
@@ -42,6 +43,18 @@ angular.module('openshiftConsole')
...
+          if(!isValid) {
+            return;
```
Comment (jwforres) [inline app/scripts/directives/nav.js]: make this return false please

**Comment by benjaminapetersen** [comment]
@jwforres swapped the implicit `return undefined` for `return false`.  
Good to go?

**Comment by jwforres** [comment]
[merge]

**Pull request merged**